tests/scripts/test_generate_data.py

```diff
@@ -126,3 +133,82 @@ def test_generate_classification_data(temporary_dir):
     assert not os.path.isfile(
         os.path.join(output_inference, "inference_1.txt")
     ), "Script generate.py should NOT generate inference_1.txt under --output dir but DID"
+
+    assert os.path.isfile(
+        os.path.join(output_header, "header.txt")
+    ), "Script generate.py should generate inference_0.txt under --output dir but did not"
+
+def test_generate_ranking_data(temporary_dir):
```

Comment:
considering your code is standardizing all 3 generation tasks (thanks!) maybe we should factorize those tests into a parametrized test. but we can leave that for another PR

Reply:
I agree